pkg/provisioner/crdprovisioner.go

```diff
@@ -166,31 +164,17 @@ func (a CachedReader) List(_ context.Context, objectList client.ObjectList, opts
 	return err
 }
 
-func NewCrdProvisioner(kubeConfig *rest.Config, objNamespace string) (*CrdProvisioner, error) {
-	azdiskClient, err := azureutils.GetAzDiskClient(kubeConfig)
-	if err != nil {
-		return nil, err
-	}
-	kubeClient, err := kubeClientset.NewForConfig(kubeConfig)
-	if err != nil {
-		return nil, err
-	}
-
-	kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, consts.DefaultInformerResync)
-	startWatchingKube(context.Background(), kubeInformerFactory)
-
-	azInformerFactory := azdiskinformers.NewSharedInformerFactory(azdiskClient, consts.DefaultInformerResync)
-
+func NewCrdProvisioner(azdiskClient azdisk.Interface, objNamespace string, cw *watcher.ConditionWatcher, kubeInformerFactory informers.SharedInformerFactory, azInformerFactory azdiskinformers.SharedInformerFactory) *CrdProvisioner {
```

Comment:
We could directly pass the `CacheReader1 instead of each parameter.

Reply:
Do you mean to create a `CacheReader` object in `azuredisk_v2.go` and pass to this function?